### PR TITLE
CSP page implementation

### PR DESCRIPTION
**Pull request by ziyue5**
Please go to "/csp" path to see the changes

### CODE REVIEW

**Review by pengyin-shan — APPROVED** [review]

File: src/components/CSP/CategoryMap.tsx [hunk 1]
```diff
@@ -8,10 +8,10 @@ import Box from "@mui/material/Box";
 import Typography from "@mui/material/Typography";
 
 import PropTypes from "prop-types";
```
Comment (ywkim312) [inline src/components/CSP/CategoryMap.tsx:10]: you're removing allstates.json but this file is still being used in EqipTotalMap maybe? Could you double check if it is safe to delete this file?

**Comment by ywkim312** [comment]
Please fix conflict

File: src/data/CSP/csp_practice_categories_data.json
Comment (sandeep-ps) [inline src/data/CSP/csp_practice_categories_data.json:1]: The CSP JSON files and the data/CSP folder can be removed, right?

Comment (pengyin-shan) [reply src/data/CSP/csp_practice_categories_data.json:1]: yes I believe, but let's leave it these and deploy to the dev, then I'll submit another PR to remove them. For safety

**Comment by pengyin-shan** [comment]
> I'm seeing a longer loading time (see screenshot below) when switching from EQIP to CSP. Going back to EQIP seems to load better. This may be more visible in slower internet speeds. Though it may not need immediate action, I wanted to share this for future improvement.
> 
> image

yes, the EQIP is having local data, which CSP has api. We should talk about this later, but let's push to dev now

File: src/pages/CSPPage.tsx [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            { name: "Structural", value: structuralTotal, color: "#CDDBD8" },
+            { name: "Bundles", value: structuralTotal, color: "#C3C5C4" }
+        ]);
+
+        setOld2014ChartData([
+            { name: "Cropland", value: croplandTotal, color: "#2F7164" },
+            { name: "Rangeland", value: rangelandTotal, color: "#4D847A" },
+            { name: "Pastureland", value: pasturelandTotal, color: "#749F97" },
+            { name: "Other: supplemental, adjustment & other", value: SAOTotal, color: "#869397" },
+            { name: "SAO", value: SAOTotal, color: "#9CBAB4" },
```
Comment (sandeep-ps) [inline src/pages/CSPPage.tsx:118]: I think this should be "Non-industrial private forestland". I'm not sure if this is part of CSP. I couldn't find that in the JSON data.

File: src/components/ProgramDrawer.tsx [hunk 3]
```diff
@@ -154,22 +154,203 @@ function EQIPCheckboxList({ setEQIPChecked, setShowPopUp }) {
...
+        "Vegetative",
+        "Forest management",
+        "Soil remediation",
+        "Structural",
+        "Bundles",
+        "Cropland",
+        "Rangeland",
+        "Pastureland",
+        "Other: supplemental, adjustment & other",
+        "NIPF",
```
Comment (sandeep-ps) [inline src/components/ProgramDrawer.tsx:181]:
```suggestion
        "Non-industrial private forestland",
```
A couple of weeks ago, Jonathan had suggested expanding the abbreviations in a direct Slack conversation related to CSP data.

File: src/data/CSP/csp_practice_categories_data.json [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    "totalPaymentInDollars": 349383709.47,
+                    "totalPaymentInPercentage": 11.8
+                },
+                {
+                    "practiceCategoryName": "Pastureland",
+                    "totalPaymentInDollars": 195464258.55,
+                    "totalPaymentInPercentage": 6.6
+                },
+                {
+                    "practiceCategoryName": "NIPF",
```
Comment (sandeep-ps) [inline src/data/CSP/csp_practice_categories_data.json:59]:
```suggestion
                    "practiceCategoryName": "Non-industrial private forestland",
```
Similar suggestion here.

File: src/data/CSP/csp_practice_categories_data.json
Comment (sandeep-ps) [reply src/data/CSP/csp_practice_categories_data.json:1]: Okay. Sounds fine. Thanks.

**Comment by sandeep-ps** [comment]
Please check if the sub-categories are working fine. When I select NIPF, no tabular data is displayed. It could be because of how it's named in the frontend vs JSON response. Similarly, with some other subcategories too.

File: src/components/ProgramDrawer.tsx (hunk 3 above)
Comment (pengyin-shan) [reply src/components/ProgramDrawer.tsx:181]: Ah I don't know this, I will change

File: src/pages/CSPPage.tsx (hunk 2 above)
Comment (pengyin-shan) [reply src/pages/CSPPage.tsx:118]: Thanks for catching, yes it should be NIPF, i'll change it

File: src/components/CSP/CategoryMap.tsx (hunk 1 above)
Comment (pengyin-shan) [reply src/components/CSP/CategoryMap.tsx]: allstates.json file is still there, in data folder. I changed allstates in csp to API. So now EQIP is using allstates.json and CSP is using allstates api.

File: src/components/ProgramDrawer.tsx (hunk 3 above)
Comment (sandeep-ps) [reply src/components/ProgramDrawer.tsx]: Thanks.

File: CHANGELOG.md [hunk 5]
```diff
@@ -20,6 +20,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Navigation from PDL title and PDL logo [#51](https://github.com/policy-design-lab/pdl-frontend/issues/51)
 - Icon for left menu [#62](https://github.com/policy-design-lab/pdl-frontend/issues/62)
 - Background color for selected columns in table [#68](https://github.com/policy-design-lab/pdl-frontend/issues/68)
+- CSP page [#74](https://github.com/policy-design-lab/pdl-frontend/issues/74)
```
Comment (sandeep-ps) [inline CHANGELOG.md:23]: Please move these to the Unreleased section at the top. Thanks.

Comment (pengyin-shan) [reply CHANGELOG.md]: pushed the change

**Comment by navarroc** [comment]
From Jonathan:

stylistic request on the menu: the order for 2018 CSP Practices should be: structural, vegetative, land management, forest management, soil remediation, existing activity, bundles and then other. for 2014 CSP Eligible Land the order should be: cropland, grassland, rangeland, pasture land, pastured cropland, nonindustrial private forestland, and other.

**Review by sandeep-ps — APPROVED** [review]
I briefly tested the pages, and the changes look fine. Thanks.

File: src/components/LandingPageMap.tsx [hunk 6]
```diff
@@ -111,7 +111,7 @@ const MapChart = ({ setTooltipContent, title }) => {
             color5 = "#045A8D";
             legendTitle = (
                 <Typography noWrap variant="h6">
-                    Total Supplemental Nutrition Assistance Programs Benefits from <strong>2018 - 2022</strong>
+                    Total Supplemental Nutrition Assistance Programs (SNAP) Costs from <strong>2018 - 2022</strong>
```
Comment (navarroc) [inline src/components/LandingPageMap.tsx:114]: Programs should be Program

**Pull request merged**